modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java

```diff
@@ -111,6 +118,24 @@ public DDMDataProviderResponse getData(
 		return DDMRESTDataProviderSettings.class;
 	}
 
+	protected String buildURL(
+		DDMDataProviderRequest ddmDataProviderRequest,
+		DDMRESTDataProviderSettings ddmRESTDataProviderSettings) {
+
+		Map<String, String> pathParameters = getPathParameters(
+			ddmDataProviderRequest, ddmRESTDataProviderSettings);
+
+		String url = ddmRESTDataProviderSettings.url();
+
+		for (Entry<String, String> pathParameter : pathParameters.entrySet()) {
+			url = StringUtil.replaceFirst(
```

Comment:
hey @rafaprax , should we replace only the first occurrence?

Reply:
@pedroqueiroz I think so. It is weird having a mapping like `api/{param1}/something/{param1}`. What do you think?

Reply:
It makes sense! :)